include/clap/ext/params.h

```diff
@@ -156,15 +156,43 @@ typedef struct clap_param_info {
    clap_param_info_flags flags;
 
    // This value is optional and set by the plugin.
-   // Its purpose is to provide a fast access to the plugin parameter:
+   // Its purpose is to provide a fast access to the
+   // plugin parameter object by caching its pointer.
+   // For instance:
    //
+   // in clap_plugin_params.get_info():
    //    Parameter *p = findParameter(param_id);
    //    param_info->cookie = p;
    //
-   //    /* and later on */
-   //    Parameter *p = (Parameter *)cookie;
+   // later, in clap_plugin.process():
    //
-   // It is invalidated on clap_host_params->rescan(CLAP_PARAM_RESCAN_ALL) and when the plugin is
+   //    Parameter *p = (Parameter *)event->cookie;
+   //    if (!p) [[unlikely]]
+   //       p = findParameter(event->param_id);
+   //
+   // where findParameter() is a function the plugin implements
+   // to map parameter ids to internal objects.
+   //
+   // The host will either provide the cookie as issued or nullptr
+   // in events addressing parameters.
+   //
+   // The plugin must gracefully handle the case of a cookie
+   // which is nullptr, but can safely assume any cookie
+   // which is not nullptr is the value it issued.
+   //
+   // The host should provide the cookie if its retrival is cheaper
+   // than an hastable lookup, otherwise the plugin may as well do
+   // the parameter lookup.
```

Comment:
I don't agree with this. The host may send multiple events on one parameter per block and will know that, the plugin will not know that unless it scans, so the host can at least compress per block. the original language also warned host developers that plugins may be substantially less performant if they skip cookies, and I think that should be restored.

So I would replace this paragraph with

"While a plugin has to handle the case of a nullptr cookie, it may have substantially decreased performance in that case, especially under heavy modulation mode. The host should make every effort to support a cookie. 

Practically, unless the host has performance worse than a hash table lookup to determine the cookie for a parameter, implementing cookies in the host will result in a more performant situation.